presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushPartialAggregationThroughExchange.java

```diff
@@ -265,6 +275,17 @@ private PlanNode split(AggregationNode node, Context context)
                 node.getGroupIdVariable());
     }
 
+    private boolean byteReductionLessThanThreshold(AggregationNode aggregationNode, ExchangeNode exchangeNode, Context context)
+    {
+        StatsProvider stats = context.getStatsProvider();
+        PlanNodeStatsEstimate exchangeStats = stats.getStats(exchangeNode);
+        PlanNodeStatsEstimate aggregationStats = stats.getStats(aggregationNode);
```

Comment:
This estimate might not be reliable, especially when aggregating on more than one column. What do you think about looking at the `exchangeStats` only? Can we extract the cardinallity for the aggregation column (if there's only a single column) and check if the cardinality above a threshold?

Reply:
Limited the scope of disabling partial aggregation to group bys on a single column